src/main/java/bisq/core/dao/blockchain/BsqBlockChain.java

```diff
@@ -345,12 +351,26 @@ public void issueBsq(TxOutput txOutput) {
 
     @Override
     public void setCreateCompensationRequestFee(long fee, int blockHeight) {
-        lock.write(() -> proposalFees.add(new Tuple2<>(fee, blockHeight)));
+        lock.write(() -> setParam(proposalFees, fee, blockHeight));
     }
 
     @Override
     public void setBlindVoteFee(long fee, int blockHeight) {
-        lock.write(() -> blindVoteFees.add(new Tuple2<>(fee, blockHeight)));
+        lock.write(() -> setParam(blindVoteFees, fee, blockHeight));
```

Comment:
These locks seem unnecessary since the lock is taken in `setParam` anyway. It doesn't change the behavior but it would be less confusing to only take the locks when necessary.

Reply:
Yes I am not happy with the current threading handling. I think I will move to using synchronized...